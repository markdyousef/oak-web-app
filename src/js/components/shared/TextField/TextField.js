// @flow
import React, { PropTypes } from 'react';
import styled from 'styled-components';
import colors from '../../../styles/colors';

const H5 = styled.h5`
    font-size: 14px;
    font-weight: lighter;
    color: ${colors.darkGrey};
    margin-bottom: 5px;
`;

const Default = styled.textarea`
    padding: 12px 8px;
    width: 100%;
    max-width: 400px;
    border-radius: 3px;
    border: 1px solid ${colors.lightGrey};
    font-size: 15px;
    border-radius: 3px;
    outline: none;
    background-color: #fff;
    resize: none;
    &:focus {

        border-color: #67C198
    }
    &:hover {
        border-color: #B5D3C9
    }
`;

const TextField = ({ value, onChange, title, placeholder }:Object) => {
    return (
        <div>
            <H5>{title}</H5>
            <Default
                value={value}
                onChange={event => onChange(event.target.value)}
                placeholder={placeholder}
            />
        </div>
    );
};

TextField.propTypes = {
    value: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    title: PropTypes.string,
    placeholder: PropTypes.string
};

TextField.defaultProps = {
    value: '',
    title: '',
    placeholder: null
};

export default TextField;
