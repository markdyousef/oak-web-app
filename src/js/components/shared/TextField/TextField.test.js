import React from 'react';
import { shallow } from 'enzyme';
import TextField from './TextField';

it('renders correctly', () => {
    const props = {
        onChange: () => {}
    }
    const shallowComponent = shallow(<TextField {...props} />);
    expect(shallowComponent).toMatchSnapshot();
});
