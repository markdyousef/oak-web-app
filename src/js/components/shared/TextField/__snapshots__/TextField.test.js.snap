// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <styled.h5 />
    <styled.textarea
        onChange={[Function]}
        placeholder={null}
        value=""
    />
</div>,
  "nodes": Array [
    <div>
      <styled.h5 />
      <styled.textarea
            onChange={[Function]}
            placeholder={null}
            value=""
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TextField
        onChange={[Function]}
        placeholder={null}
        title=""
        value=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onChange": [Function],
          "placeholder": null,
          "title": "",
          "value": "",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <styled.h5 />
          <styled.textarea
                    onChange={[Function]}
                    placeholder={null}
                    value=""
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <styled.h5 />
          <styled.textarea
                    onChange={[Function]}
                    placeholder={null}
                    value=""
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TextField
    onChange={[Function]}
    placeholder={null}
    title=""
    value=""
/>,
}
`;
